lib/routes/thegadgetflow/rss.ts

```diff
@@ -0,0 +1,100 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import { art } from '@/utils/render';
+import path from 'node:path';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+
+export const route: Route = {
+    path: '/:category?',
+    categories: ['shopping'],
+    example: '/thegadgetflow/cool-gadgets-gifts',
+    parameters: { category: 'category name, can be found in url' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['thegadgetflow.com/:category'],
+            target: '/:category',
+        },
+    ],
+    name: 'Category',
+    maintainers: ['EthanWng97'],
+    handler,
+};
+
+async function handler(ctx) {
+    const baseUrl = 'https://thegadgetflow.com';
+    const categoryApiPath = '/wp-json/wp/v2/categories';
+    const postApiPath = '/wp-json/wp/v2/posts';
+
+    // get category number
+    const categorySlug = ctx.req.param('category') || '';
+
+    let categoryId;
+    let categoryResponse;
+    if (categorySlug) {
+        categoryResponse = await cache.tryGet(`${baseUrl}${categoryApiPath}`, async () => {
+            const { data } = await got(`${baseUrl}${categoryApiPath}`, {
+                searchParams: { slug: categorySlug },
+            });
+            return data;
+        });
+        if (!categoryResponse || categoryResponse.length === 0) {
+            throw new Error(`Category "${categorySlug}" not found`);
+        }
```

Comment:
Perform the length check inside the cache and cache [0] instead of the whole array